src/fparser/Fortran2003.py

```diff
@@ -5506,42 +5532,51 @@ def match(string):
         if not l: return
         return Format(repmap(line[:i].rstrip())), Output_Item_List(l)
     match = staticmethod(match)
+
     def tostr(self):
         if self.items[1] is None: return 'PRINT %s' % (self.items[0])
         return 'PRINT %s, %s' % tuple(self.items)
 
+
 class Io_Control_Spec_List(SequenceBase): # R913-list
     """
     <io-control-spec-list> is a list taking into account C910, C917, C918
     """
     subclass_names = []
     use_names = ['Io_Control_Spec']
+    @staticmethod
     def match(string):
         line, repmap = string_replace_map(string)
         splitted = line.split(',')
-        if not splitted: return
         lst = []
-        for i in range(len(splitted)):
-            p = splitted[i].strip()
-            if i==0:
-                if '=' not in p: p = 'UNIT=%s' % (repmap(p))
-                else: p = repmap(p)
-            elif i==1:
-                if '=' not in p:
-                    p = repmap(p)
-                    try:
-                        f = Format(p)
-                        # todo: make sure that f is char-expr, if not, raise NoMatchError
-                        p = 'FMT=%s' % (Format(p))
-                    except NoMatchError:
-                        p = 'NML=%s' % (Namelist_Group_Name(p))
-                else:
-                    p = repmap(p)
+        unit_is_positional = False
+        for idx in range(len(splitted)):
+            spec = splitted[idx].strip()
+            spec = repmap(spec)
+            if idx == 0 and "=" not in spec:
+                # Must be a unit number. However, we do not prepend "UNIT="
+                # to it in case the following Io_Control_Spec is positional
+                # (and therefore either a Format or Namelist spec).
+                lst.append(Io_Control_Spec(spec))
+                unit_is_positional = True
+            elif idx == 1 and "=" not in spec:
+                if not unit_is_positional:
+                    # Cannot have a positional argument following a
+                    # named argument
+                    return
+                # Without knowing the type of the variable named in spec
+                # we have no way of knowing whether this is a format or
+                # a namelist specifier. However, if it is a character
+                # constant or "*" then it must be a Format spec and we can
+                # prepend "FMT=" to it.
+                spec = spec.lstrip().rstrip()
+                if Char_Literal_Constant.match(spec) or \
```

Comment:
I'm following what is done currently which is to mark-up each argument with a name where possible. In general this is not possible for un-named format or namelist arguments (which is the reason for this issue). However, I think it is best to provide as much information as we can so if we can determine that a particular argument is a format specifier, then we should specify that it is.